src/nnet3/nnet-chain-example.cc

```diff
@@ -31,30 +31,20 @@ void NnetChainSupervision::Write(std::ostream &os, bool binary) const {
   WriteToken(os, binary, name);
   WriteIndexVector(os, binary, indexes);
   supervision.Write(os, binary);
-  WriteToken(os, binary, "<DW>");  // for DerivWeights.  Want to save space.
-  WriteVectorAsChar(os, binary, deriv_weights);
   WriteToken(os, binary, "</NnetChainSup>");
 }
 
 bool NnetChainSupervision::operator == (const NnetChainSupervision &other) const {
   return name == other.name && indexes == other.indexes &&
-      supervision == other.supervision &&
-      deriv_weights.ApproxEqual(other.deriv_weights);
+      supervision == other.supervision;
 }
 
 void NnetChainSupervision::Read(std::istream &is, bool binary) {
   ExpectToken(is, binary, "<NnetChainSup>");
   ReadToken(is, binary, &name);
   ReadIndexVector(is, binary, &indexes);
   supervision.Read(is, binary);
-  std::string token;
-  ReadToken(is, binary, &token);
-  // in the future this back-compatibility code can be reworked.
-  if (token != "</NnetChainSup>") {
```

Comment:
You'll need to keep the code that reads the deriv-weights, but just do nothing with them.  Otherwise the code will crash when running on previously dumped egs.